### PR TITLE
Add new connectivity overview page

### PR DESCRIPTION
**Pull request by pipo02mix**
### What this PR does / why we need it

Towards https://github.com/giantswarm/giantswarm/issues/30732

### Things to check/remember before submitting

- If you made content changes

    - Run `make dev` to render and proofread content changes locally.
    - Bump `last_review_date` in the front matter header if you reviewed the entire page.

### CODE REVIEW

**Review by AndiDog — COMMENTED** [review]
I'm not sure how to review this. I guess we should be more specific already in this draft, and therefore rather pull in Shield's docs champion to outline this overview draft?

**Comment by pipo02mix** [comment]
It is a draft, I just marked it now. I have created a ticket to explain what is the purpose of this doc entry. I am hesitating if we need just an outline or we (as sig docs participants) can further facilitate the article's writing. Now the body contains the template outline and some suggestions on how to write it. My thinking was to let the champion write the draft and help him later with wording and syntax to be consistent and similar to other sibling pages. WDYT?

File: src/content/overview/connectivity/_index.md [hunk 1]
```diff
@@ -10,3 +10,26 @@ last_review_date: 2024-03-18
...
+- public/private networking
+- CNI
+- Policies
+- Ingress
+- API gateway
+- Service mesh
+- DNS
+- egress proxy
+
+Write a brief description of the the connectivity in general from introductory point of view.
```
Comment (pipo02mix) [inline src/content/overview/connectivity/_index.md:27]:
```suggestion
The main idea is to write a brief description of connectivity and networking in general from an introductory point of view setting the high-level goals we would like to achieve.
```

**Comment by AndiDog** [comment]
> It is a draft, I just marked it now. I have created a ticket to explain what is the purpose of this doc entry. I am hesitating if we need just an outline or we (as sig docs participants) can further facilitate the article's writing. Now the body contains the template outline and some suggestions on how to write it. My thinking was to let the champion write the draft and help him later with wording and syntax to be consistent and similar to other sibling pages. WDYT?

I think SIG Docs _could_ provide an outline, but then again the champion is the expert of what should even be included in the article (whether it's in "Overview" or "Getting started", doesn't matter). Getting together online to draft an outline could work, instead of trying that asynchronously (which might be slower).

File: src/content/overview/connectivity/_index.md [hunk 2]
```diff
@@ -6,7 +6,38 @@ menu:
...
+- **Ingress traffic management**: manage your ingress traffic using an API gateway, ingress controllers, or native load balancers to ensure reliable and efficient routing of external traffic to your services.
+- **Egress proxy add-on**: safeguard your outbound traffic with an egress proxy. Our system allows to plug in your own proxy for controlling the egress traffic from your clusters.
+- **Scalable container network**: the container network layer relies on Cilium, a high-performance container network interface that provides secure communication between your pods and scales reliably with your cluster.
+- **Internal DNS**: the platform comes with CoreDNS as DNS solution, which provides a flexible and scalable DNS server for efficient service discovery within your clusters.
+- **Encrypted traffic**: secure your communication with mTLS (mutual Transport Layer Security) encryption, ensuring that all traffic between services is encrypted.
+- **Traffic management**: manage traffic routing within your clusters using service mesh technologies like Linkerd, which provide advanced features like circuit breakers, retries, timeouts, and rate limiting.
```
Comment (weatherhog) [inline src/content/overview/connectivity/_index.md:29]: I am not sure if we should keep it this way as we are not actively supporting service Mesh at the moment?

File: src/content/overview/connectivity/_index.md [hunk 3]
```diff
@@ -6,7 +6,38 @@ menu:
...
+
+## Cloud-native technologies
+
+Our platform leverages various projects under the Cloud Native initiative that help us offer aforementioned capabilities:
+
+- **Kubernetes**: As the backbone of our platform, Kubernetes give robust orchestration capabilities for managing containerized workloads and services.
+- **Cilium**: This container network interface provides secure communication between your pods and scales reliably with your cluster.
+- **Nginx**: Ingrees controller that helps in managing your ingress traffic efficiently, ensuring reliable routing of external traffic to your services.
+- **CoreDNS**: Flexible DNS server improves service discovery within your clusters, aiding in efficient internal DNS management.
+- **Linkerd**: As leading service mesh options, they enhance the security, reliability, and observability of microservices communication within your clusters.
```
Comment (weatherhog) [inline src/content/overview/connectivity/_index.md:40]: same here I think we should remove Linkerd

File: src/content/overview/connectivity/_index.md [hunk 4]
```diff
@@ -6,7 +6,38 @@ menu:
...
+- **Traffic management**: manage traffic routing within your clusters using service mesh technologies like Linkerd, which provide advanced features like circuit breakers, retries, timeouts, and rate limiting.
+- **Resilience:**:It improves resilience by providing features such as circuit breakers, retries, timeouts, and rate limiting. These help in mitigating issues related to network failures and service overloads.
+
+## Cloud-native technologies
+
+Our platform leverages various projects under the Cloud Native initiative that help us offer aforementioned capabilities:
+
+- **Kubernetes**: As the backbone of our platform, Kubernetes give robust orchestration capabilities for managing containerized workloads and services.
+- **Cilium**: This container network interface provides secure communication between your pods and scales reliably with your cluster.
+- **Nginx**: Ingrees controller that helps in managing your ingress traffic efficiently, ensuring reliable routing of external traffic to your services.
```
Comment (weatherhog) [inline src/content/overview/connectivity/_index.md:38]: Type here Ingrees instead of Ingress

**Review by weatherhog — CHANGES_REQUESTED** [review]
Maybe we should also add k8s-node-local-dns, external DNS and Kong? What do you think?

**Review by weatherhog — APPROVED** [review]

**Pull request merged**